src/refactor_nrepl/ns/clean_ns.clj

```diff
@@ -35,20 +36,23 @@
   (assert-no-exclude-clause (core/get-ns-component ns-form :use))
   ns-form)
 
-(defn clean-ns [{:keys [path]}]
-  {:pre [(seq path) (string? path) (core/source-file? path)]}
-  ;; Prefix notation not supported in cljs.
-  ;; We also turn it off for cljc for reasons of symmetry
-  (config/with-config {:prefix-rewriting (if (or (core/cljs-file? path)
-                                                 (core/cljc-file? path))
-                                           false
-                                           (:prefix-rewriting config/*config*))}
-    (let [ns-form (validate (core/read-ns-form-with-meta path))
-          deps-preprocessor (if (get config/*config* :prune-ns-form)
-                              #(prune-dependencies % path)
-                              identity)
-          new-ns-form (-> (ns-parser/parse-ns path)
-                          deps-preprocessor
-                          (rebuild-ns-form ns-form))]
-      (when-not (= ns-form new-ns-form)
-        new-ns-form))))
+(defn clean-ns [{:keys [path relative-path]}]
+  {:pre [(seq path) (string? path)]}
+  ;; Try first the absolute, then the relative path
+  (let [path (first (filter #(some-> % io/file .exists) [path relative-path]))]
```

Comment:
`some` might be better here since we're looking for the first thing that fulfills some `pred`

Reply:
Since the predicate returns boolean rather than the truthy value I find it awkward to use `some` for such cases, as you end up with `(some #(and (pred? %) %) coll)`. Would you still prefer `some`?